client/task_runner.go

```diff
@@ -1176,8 +1176,8 @@ func (r *TaskRunner) startTask() error {
 
 		r.logger.Printf("[WARN] client: error from prestart: %v", wrapped)
 
-		if rerr, ok := err.(*structs.RecoverableError); ok {
-			return structs.NewRecoverableError(wrapped, rerr.Recoverable)
+		if rerr, ok := err.(structs.Recoverable); ok {
+			return structs.NewRecoverableError(wrapped, rerr.Recoverable())
```

Comment:
It honestly might be worth adding a helper function.

`func NewWrappedRecoverableError(wrapped, original) { ... }`

It just checks if the original is recoverable.